### PR TITLE
Query multiple tables

### PR DESCRIPTION
**Pull request by estebanmual**
In this milestone, I:
  - Used primary key & foreign key mechanisms for joining tables.
  - Queried multiple tables.
  - Prepared complex queries that answer analytical questions.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
+    name TEXT
+);
+
+-- Create animal table
+CREATE TABLE animals
+(
+    animal_id INTEGER PRIMARY KEY,
     name TEXT NOT NULL,
     date_of_birth DATETIME,
     escape_attempts INTEGER,
```
Comment (Gambit142) [inline schema.sql]: - [x] Kindly change the data type for `date_of_birth` to `DATE`. There is no data type as `DATETIME` in PostgreSQL. 👍

File: data.sql [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
+    ('Pikachu', '2021-01-07', 15.04, 'false', 1, 1, 2),
+    ('Devimon', '2017-05-12', 11, 'true', 5, 2, 3),
+    ('Charmander', '2020-02-08', -11, 'false', 0, 1, 4),
+    ('Plantmon', '2021-11-15', -5.7, 'true', 2, 2, 3),
+    ('Squirtle', '1993-05-12', -12.13, 'false', 3, 1, 4),
+    ('Angemon', '2005-06-12', -45, 'true', 1, 2, 5),
+    ('Boarmon', '2005-06-7', 20.4, 'true', 7, 2, 5),
+    ('Blossom', '1998-10-13', 17, 'true', 3, 1, 4),
+    ('Ditto', '2022-05-14', 22, 'true', 4, 1)
+
```
Comment (Gambit142) [inline data.sql]: - [x] The data entry for _**Ditto**_ is not complete. It is missing an entry. Kindly fix it. 👍

File: schema.sql [hunk 3]
```diff
@@ -1,11 +1,34 @@
-/* Database schema to keep the structure of entire database. */
+-- These scripts will delete the tables if it's already there
+DROP TABLE IF EXISTS owners;
+DROP TABLE IF EXISTS species;
+DROP TABLE IF EXISTS animals;
```
Comment (Gambit142) [inline schema.sql:4]: - [x] (Optional) I would advise you to drop tables and their dependents whenever you want to recreate the table. To do this, you need to use the key work _**CASCADE**_. `DROP TABLE IF EXISTS owners CASCADE`. 👍

File: schema.sql [hunk 4]
```diff
@@ -1,11 +1,34 @@
...
 
-CREATE TABLE animals (
-    id INTEGER PRIMARY KEY,
+-- Create owners table
+CREATE TABLE owners
+(
+	owner_id INTEGER PRIMARY KEY,
```
Comment (Gambit142) [inline schema.sql]: - [x] When you make a column like this for `owner_id`, you will have to manually set the value in your _**data.sql**_ file. I would advise you to make it automatic. You could do this `owner_id SERIAL PRIMARY KEY`. The reason I asked you for this change is that when I run the **_data.sql file_**, I get this error:

image

File: schema.sql [hunk 5]
```diff
@@ -1,11 +1,34 @@
...
+	full_name TEXT NOT NULL,
+	age INTEGER
+);
+
+-- Create species table
+CREATE TABLE species
+(
+    specie_id INTEGER PRIMARY KEY,
```
Comment (Gambit142) [inline schema.sql]: - [x] Kindly refer to the statement concerning setting the `id` for a table. 👍

File: schema.sql [hunk 6]
```diff
@@ -1,11 +1,34 @@
...
+    name TEXT
+);
+
+-- Create animal table
+CREATE TABLE animals
+(
+    animal_id INTEGER PRIMARY KEY,
```
Comment (Gambit142) [inline schema.sql]: - [x] Kindly refer to the statement concerning setting the `id` for a table. 👍

File: queries.sql [hunk 7]
```diff
@@ -70,3 +70,17 @@ SELECT species, AVG(escape_attempts)
...
+INNER JOIN owners o
+	ON a.owner_id = o.owner_id
+WHERE o.full_name = 'Melody Pond';
+
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT a.name
+FROM animals a
+INNER JOIN species s
+	ON s.specie_id = a.specie_id
+WHERE s.name = 'Pokemon';
```
Comment (Gambit142) [inline queries.sql]: - [x] Great work writing the first two queries, kindly add queries for the following:

- List all owners and their animals, remember to include those that don't own any animal.
- How many animals are there per species?
- List all Digimon owned by Jennifer Orwell.
- List all animals owned by Dean Winchester that haven't tried to escape.
- Who owns the most animals?

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @estebanmual 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- _Kindly ensure that you test the application before submitting it._ 👍 
- _Kindly check the comments under the review for all required changes._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- You can also consider the optional change I made under the review. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: data.sql (hunk 2 above)
Comment (estebanmual) [reply data.sql]: Hi @Gambit142, How can I declare Ditto does not have any owner?

**Review by JuliCarracedo — APPROVED** [review]
# Status: Approved ✔️
Hi @estebanmual , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**